### PR TITLE
fix proxy and tls

### LINKED ISSUE #7355: insecure_skip_tls_verify isn't working
Hey, 
I'm trying to use insecure_skip_tls_verify option and it doesn't seem to work.
it does't work since 1.3.4 release, probably because of #7226.
```
/ # packer build --debug scb.json
Debug mode enabled. Builds will not be parallelized.
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name: SCB-CNF-20190226-0935
==> amazon-ebs: Error querying AMI: RequestError: send request failed
==> amazon-ebs: caused by: Post https://10.16.146.93/api/v2/aws/ec2/: x509: cannot validate certificate for 10.16.146.93 because it doesn't contain any IP SANs
Build 'amazon-ebs' errored: Error querying AMI: RequestError: send request failed
caused by: Post https://10.16.146.93/api/v2/aws/ec2/: x509: cannot validate certificate for 10.16.146.93 because it doesn't contain any IP SANs
```
/ # packer version
Packer v1.3.4

example json:
```
  "builders": [
    {
      "type": "amazon-ebs",
      "region": "symphony",
      "availability_zone": "symphony",
      "skip_region_validation": true,
      "custom_endpoint_ec2": "https://10.16.146.93/api/v2/aws/ec2",
      "insecure_skip_tls_verify": true,

```
  
cc: @azr, @gmmephisto

### PR DESCRIPTION
**Pull request by SwampDragons**
When I made the code change to force http proxy, the tls_verification, as well as some other custom connection config code, broke. This should fix it all.

Closes #7355

### CODE REVIEW

**Comment by SwampDragons** [comment]
Linux:
[packer.zip](https://github.com/hashicorp/packer/files/2921558/packer.zip)

osx:
[packer.zip](https://github.com/hashicorp/packer/files/2921561/packer.zip)

windows:
[packer.zip](https://github.com/hashicorp/packer/files/2921563/packer.zip)

**Comment by eran-stratoscale** [comment]
Hey, I tried using the zip you added, It doesn't seems to work:
```
./packer build packer.json
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name: SCB-CNF-20190226-0935
==> amazon-ebs: Error querying AMI: RequestError: send request failed
==> amazon-ebs: caused by: Post https://10.16.145.59/api/v2/aws/ec2/: x509: cannot validate certificate for 10.16.145.59 because it doesn't contain any IP SANs
Build 'amazon-ebs' errored: Error querying AMI: RequestError: send request failed
caused by: Post https://10.16.145.59/api/v2/aws/ec2/: x509: cannot validate certificate for 10.16.145.59 because it doesn't contain any IP SANs

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Error querying AMI: RequestError: send request failed
caused by: Post https://10.16.145.59/api/v2/aws/ec2/: x509: cannot validate certificate for 10.16.145.59 because it doesn't contain any IP SANs
```
Tried building and running your branch myself, also didn't help.

File: builder/amazon/common/access_config.go [hunk 1]
```diff
@@ -210,9 +210,7 @@ func (c *AccessConfig) NewEC2Connection() (ec2iface.EC2API, error) {
 		return nil, err
 	}
 
-	ec2conn := ec2.New(sess, &aws.Config{
-		HTTPClient: commonhelper.HttpClientWithEnvironmentProxy(),
```
Comment (azr) [inline builder/amazon/common/access_config.go]: I think we might still need that client from the commonhelper.

Comment (SwampDragons) [reply builder/amazon/common/access_config.go]: No I think that was my original mistake -- I believe the config we're defining up in line 74 where I changed the code here is being attached to session via the session.Options, so it's already in there when we're calling ec2.New. I think what was happening here is I was overwriting the configuration that we did on the session, and that broke everything _but_ the proxy. I set up a little toy proxy using tinyProxy with this PR and it worked, so I consider that validation that we don't need that commonhelper tool I wrote.

**Comment by SwampDragons** [comment]
Thanks for testing this. I think I found the missing line. Can you try again now?

Turns out I was overriting this in the access config and inside the builder. I'm going to go back and clean up the code further but what I have pushed should be enough to get this working again.

File: builder/amazon/common/access_config.go (hunk 1 above)
Comment (azr) [reply builder/amazon/common/access_config.go]: ah, nice ! thanks for the explanation :)

**Comment by eran-stratoscale** [comment]
Works!
Thank you.

**Pull request merged**